sinch/core/signature.py

```diff
@@ -0,0 +1,45 @@
+import hashlib
+import hmac
+import base64
+from datetime import datetime, timezone
+import json
+
+
+class Signature:
+    def __init__(self, sinch, http_method, request_data, request_uri):
+        self.sinch = sinch
+        self.http_method = http_method
+        self.content_type = 'application/json; charset=UTF-8'
+        self.request_data = request_data
+        self.signature_time = datetime.now(timezone.utc).isoformat()
+        self.request_uri = request_uri
+        self.authorization_signature = None
+
+    def get_http_headers_with_signature(self):
+        return {
+            "Content-Type": self.content_type,
+            "Authorization": (
+                f"Application {self.sinch.configuration.verification_key}:{self.authorization_signature}"
+            ),
+            "x-timestamp": self.signature_time
+        }
+
+    def calculate(self):
+        b64_encoded_application_secret = base64.b64decode(self.sinch.configuration.verification_secret)
+        encoded_verification_request = json.dumps(self.request_data).encode()
```

Comment:
Can we imagine receiving `None` for self.request_data ? (I have in mind a GET request without payload)
If yes, is encoding such case if still correct ?

Reply:
I'm working on it :)

Reply:
fixed